internal/streams/streams.go

```diff
@@ -61,6 +61,9 @@ func New(name string, source string) *Stream {
 		return nil
 	}
 
+	streamsMu.Lock()
```

Comment:
Better to use:

```go
stream := NewStream(source)
streamsMu.Lock()
streams[name] = stream
streamsMu.Unlock()
return stream
```

Reply:
in golang style is better to defer unlock

```golang
stream := NewStream(source)
streamsMu.Lock()
defer streamsMu.Unlock()
streams[name] = stream
return stream
```

Reply:
No. Defer should be used wisely where it's needed. In this case, there is no reason to use it.